src/applications/vaos/appointment-list/redux/actions.js

```diff
@@ -730,3 +731,27 @@ export function updateBreadcrumb(breadcrumb) {
     dispatch({ type: UPDATE_BREADCRUMB, breadcrumb });
   };
 }
+/**
+ * Function to retrieve provider information from the provider
+ * endpoint when using the v2 api.
+ *
+ * @param {*} appointments
+ */
+export function getProviderInfoV2(appointments) {
+  // Provider information included with v2 provider api call.
+  return async (dispatch, getState) => {
+    const featureVAOSServiceCCAppointments = selectFeatureVAOSServiceCCAppointments(
+      getState(),
+    );
+    if (featureVAOSServiceCCAppointments) {
+      const ProviderNpi = getPreferredCCProviderNPI(appointments);
+
+      const providerData = await getTransformedPreferredProvider(ProviderNpi);
+
+      dispatch({
+        type: FETCH_PROVIDER_SUCCEEDED,
+        providerData,
+      });
+    }
```

Comment:
Add a condition to dispatch providerData to prevent using what's cached in state if user decides to make another CC request but chooses NOT to select a provider after first having created a CC request with a specific provider.
```suggestion
      dispatch({
        type: FETCH_PROVIDER_SUCCEEDED,
        providerData,
      });
    }
    if (
    featureVAOSServiceCCAppointments &&
    appointments.practitioners.length === 0
    ) {
      dispatch({
         type: FETCH_PROVIDER_SUCCEEDED,
         providerData: null,
      });
   }
```